pkg/controller/v1alpha2/applicationconfiguration/component.go

```diff
@@ -173,6 +184,15 @@ func (c *ComponentHandler) createControllerRevision(mt metav1.Object, obj runtim
 		c.l.Info(fmt.Sprintf("error create controllerRevision %v", err), "componentName", mt.GetName())
 		return false
 	}
+
+	if int64(c.defaultRevisionLimit) < nextRevision {
+		go func(mt metav1.Object, curComp *v1alpha2.Component) {
```

Comment:
If user modify component many times in a short period, I'm afraid we will lose control here.  More general way is setup only one goroutine to do GC, and use a channel to receive event triggered here.

Reply:
Great!